packages/hatchify-core/src/types/schema.ts

```diff
@@ -0,0 +1,17 @@
+export interface Relationship {
+  target: string
+  options: {
+    as: string
+  }
+}
+
+export interface Schema {
+  name: string
+  attributes: {
+    [field: string]: string
+  }
+  hasOne?: Relationship[]
```

Comment:
for the old schema yes, the idea is if a `User` has one `Occupation` and one `Partner` it would look like:
```
hasOne: [ occupationRelationship, partnerRelationship ]
```
this is all subject to change, but we're just accommodating the old backend schema type for now